component/web/security/src/main/java/org/exoplatform/web/login/recovery/PasswordRecoveryHandler.java

```diff
@@ -132,11 +140,8 @@ public boolean execute(ControllerContext context) throws Exception {
                     message = message.replace("{0}", username);
                     errors.add(message);
                 } else {
-                    if (password == null || password.isEmpty() || password.length() < 6 || password.length() > 30) {
-                        errors.add(bundle.getString("gatein.forgotPassword.invalidPassword"));
-                    }
-                    if (confirmPass == null || confirmPass.length() < 6 || confirmPass.length() > 30) {
-                        errors.add(bundle.getString("gatein.forgotPassword.invalidConfirmPassword"));
+                  if (!expected(validator,uiFormInput)) {
```

Comment:
No need to recreate the UIFormStringInput , you can use the password and validate it.

Reply:
Done!!!